controller/linodemachine_controller.go

```diff
@@ -622,13 +637,16 @@ func (r *LinodeMachineReconciler) reconcileUpdate(
 			machineScope.LinodeMachine.Spec.ProviderID = nil
 			machineScope.LinodeMachine.Spec.InstanceID = nil
 			machineScope.LinodeMachine.Status.InstanceState = nil
+			machineScope.LinodeMachine.Status.Conditions = nil
 
 			conditions.MarkFalse(machineScope.LinodeMachine, clusterv1.ReadyCondition, "missing", clusterv1.ConditionSeverityWarning, "instance not found")
 		}
-
+		if err = r.removeMachineFromLB(ctx, logger, machineScope); err != nil {
+			logger.Error(err, "Failed to remove machine from LB")
+			return res, nil, err
+		}
```

Comment:
Keep the error scoped to the `if` block. It will make any future refactoring much easier.

```suggestion
		if err := r.removeMachineFromLB(ctx, logger, machineScope); err != nil {
			logger.Error(err, "Failed to remove machine from LB")
			return res, nil, err
		}
```

Reply:
done